### ISSUE TITLE
Saga correlation on string property returns DB error when value is `null`

### ISSUE THREAD

**Issue by ramonsmits** [Particular/NServiceBus.Persistence.Sql#361]
If the correlation value of a string property is `null` the following error is raised:
```txt
System.Data.SqlClient.SqlException (0x80131904): The parameterized query '(@propertyValue nvarchar(4000))
select
    Id,
    SagaTypeVe' expects the parameter '@propertyValue', which was not supplied.
```

It is very unclear what the cause for this is.

The saga persister should not at all go to the database when the correlation value is `null` but return a meaningful error.

**Comment by aleksandr-samila** [comment]
I run this test:
![image](https://user-images.githubusercontent.com/5980232/107287036-0d785680-6a6a-11eb-824d-55d8c7f08759.png)
And the current message is:
`...+SagaWithCorrelationPropertyEndpoint+SagaWithCorrelatedProperty has attempted to assign null to the correlation property CorrelatedProperty. Correlation properties cannot be assigned null.`

Can this be counted as a meaningful exception?

**Comment by mikeminutillo** [comment]
With that error message, we are fine to close this issue. It looks like core is catching it before the persistence has a chance to get involved.

**Comment by ramonsmits** [comment]
@aleksandr-samila @mikeminutillo Too bad this was sufficient to close the issue. I did some additional digging:

This test was added via https://github.com/Particular/NServiceBus/pull/5442 and seems that this was part of NServiceBus 7.2.0, meaning this is fixed since 7.2.0 or later.
